staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2149,6 +2149,33 @@ const (
 	PullIfNotPresent PullPolicy = "IfNotPresent"
 )
 
+// ContainerResizePolicy lets users specify how to handle container resize.
+// Only one of the following container resize policies may be specified.
+// If not specified, it defaults to 'RestartNotRequired'.
+type ContainerResizePolicy string
+
+// These are the valid container resize policy values:
+// 'RestartNotRequired' (default) tells Kubelet to call UpdateContainerResources
+// CRI API to resize the resources without restarting the container, if possible.
+// 'Restart' tells Kubelet to stop and start the container with when new resources
+// are applied. This is needed for legacy applications e.g. java apps using -xmxN
+// flag which are unable to use the resized memory without restarting.
+const (
+	// Resize the container in-place without restarting it.
+	RestartNotRequired ContainerResizePolicy = "RestartNotRequired"
+	// Resize the container in-place by restarting it.
+	Restart ContainerResizePolicy = "Restart"
+)
+
+// ResizePolicy represents the resource resize policy for a single container.
+type ResizePolicy struct {
```

Comment:
I would rename the current `ContainerResizePolicy` to `ResourceResizePolicy` and then rename this to `ContainerResizePolicy`

Reply:
done in upcoming commit.

Reply:
fixed in commit https://github.com/kubernetes/kubernetes/pull/102884/commits/b60c5d045bb85eacae064e209994c4afe22a4e5f